components/locale-switcher/locale-switcher.tsx

```diff
@@ -0,0 +1,88 @@
+'use client';
+
+import React from 'react';
+import { usePathname } from 'next/navigation';
+import Image from 'next/image';
+import Link from 'next/link';
+
+import './locale-switcher.scss';
+
+import { i18n, Locale } from '@/i18n-config';
+import ChevronDownIcon from '@/components/icons/chevron-down-icon';
+
+const languages: Record<Locale, string> = {
+  ca: 'Català',
+  en: 'English',
+};
+
+type Props = {
+  locale: Locale;
+};
+
+const LocaleSwitcher = ({ locale }: Props) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const pathName = usePathname();
+  const redirectedPathName = (locale: string) => {
+    if (!pathName) return '/';
+    const segments = pathName.split('/');
+    segments[1] = locale;
+    return segments.join('/');
+  };
+
+  return (
+    <div className="locale-switcher absolute right-0 top-0 z-20">
+      <div className="flex gap-4 flex flex-col items-end p-4">
+        <div
+          className={`locale-switcher-anchor relative cursor-pointer rounded-full border border-neutral-800 overflow-hidden transition-transform hover:scale-110 ${
+            isOpen ? 'opened' : ''
+          }`}
+          onClick={() => setIsOpen((o) => !o)}
+          style={{ width: 24, height: 24 }}
+        >
+          <Image
+            priority
+            src={`/assets/flags/${locale}.svg`}
+            height={24}
+            width={24}
+            alt={languages[locale]}
+            className="rounded-full"
+          />
+          <div className="locale-switcher-anchor-arrow absolute top-0 w-full h-full flex justify-center items-center">
+            <ChevronDownIcon className="mt-[2px]" />
+          </div>
+        </div>
+      </div>
+
+      <ul
+        className={`mt-[-8px] pr-4 flex gap-1 flex-col items-end ${
+          isOpen ? 'opened' : ''
+        }`}
```

Comment:
És el mateix, però també es podria fer amb la lib clsx (que la tenim de dependencia):
```jsx
<ul className={clsx('mt-[-8px] pr-4 flex gap-1 flex-col items-end', isOpen && 'opened')} />
```

Up to you 👌